CMakeLists.txt

```diff
@@ -287,7 +287,7 @@ endforeach()
 set(includedirs variants/${CONFIG_ARDUINO_VARIANT}/ cores/esp32/ ${ARDUINO_LIBRARIES_INCLUDEDIRS})
 set(srcs ${CORE_SRCS} ${ARDUINO_LIBRARIES_SRCS})
 set(priv_includes cores/esp32/libb64)
-set(requires spi_flash esp_partition mbedtls wifi_provisioning wpa_supplicant esp_adc esp_eth http_parser)
+set(requires spi_flash esp_partition mbedtls wifi_provisioning wpa_supplicant esp_adc esp_eth http_parser esp_psram)
```

Comment:
Should it rather be in priv_requires? Seems like the public headers of arduino-esp32 don't include headers from esp_psram.

Reply:
I'm just not sure if users will not need to include anything from it

Reply:
In that case they should add `esp_psram` to the private requires list of their component. `idf.py` will also show a build-time hint if they miss this dependency and try to use the header file.

You only need to add components which provide header files which arduino-esp32 public headers include. If something is only used in a .c/.cpp file, or in a private header file, then it is sufficient to add that to the private requirements, instead.

(Of course, removing things from public requirements list is a breaking change, but since esp_psram wasn't there to begin with, I think it can go into the private requirements list.)

Reply:
@igrr - This also happens when using Arduino as IDF component. Maybe that fix that you and @me-no-dev were working to add a sort of ARDUINO_LIB() to the project/user CMakeLists.txt could help.

Is it something that you guys are still working on?

Reply:
No, I don't think the issue discussed here is related to CMake helper for registering Arduino libraries. The issue here is simply that the build system didn't include esp_psram component into the build, because there was no component which added it as a requirement.